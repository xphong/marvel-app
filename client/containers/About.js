import React from 'react';

const About = () => (
  <div className="fadeIn-page-animation">
    <h1>About</h1>
    <h4>Marvel Search App</h4>
    <ul>
      <li>Searches the Marvel database for the characters, power levels</li>
      <li>Built using <a href="https://github.com/xphong/simple-react-redux-starter" target="_blank" rel="noopener noreferrer">simple-react-redux-starter</a> and the <a href="http://developer.marvel.com/" target="_blank" rel="noopener noreferrer">Marvel API</a></li>
      <li><a href="https://github.com/xphong/marvel-react-app" target="_blank" rel="noopener noreferrer">Github Link</a></li>
      <li><a href="http://marvel.com" target="_blank" rel="noopener noreferrer">Data provided by Marvel. Â© {new Date().getFullYear()} Marvel</a></li>
    </ul>
  </div>
);

export default About;
