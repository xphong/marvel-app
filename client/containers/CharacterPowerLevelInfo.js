import PropTypes from 'prop-types';
import React, { Component } from 'react';

import CharacterPowerLevelInfoDisplay from '../components/characterpowerlevels/CharacterPowerLevelInfoDisplay';
import CharacterPowerLevelRadarChart from '../components/characterpowerlevels/CharacterPowerLevelRadarChart';
import CharacterPowerLevelBarChart from '../components/characterpowerlevels/CharacterPowerLevelBarChart';

export default class CharacterPowerLevelInfo extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    const labels = ['Intelligence', 'Strength', 'Speed', 'Durability', 'Energy Projection', 'Fighting Ability'];
    const label = 'Character Data Set';
    const backgroundColor = ['#FF6384', '#4BC0C0', '#FFCE56', '#E7E9ED', '#36A2EB'];
    const data = [];

    data.push(character.Intelligence);
    data.push(character.Strength);
    data.push(character.Speed);
    data.push(character.Durability);
    data.push(character.Energy_Projection);
    data.push(character.Fighting_Ability);

    const characterChartData = {
      labels,
      datasets: [
        {
          label,
          backgroundColor,
          data
        }
      ]
    };

    return characterChartData;
  }

  render() {
    const {data} = this.props.powerLevels;
    const {id} = this.props;
    const character = data[id];

    if (!character) return null;

    const characterChartData = this.getCharacterChartData(character);

    return (
      <div className='ui container center aligned'>
        <CharacterPowerLevelInfoDisplay character={character} />
        <div className='chart-area'>
          <CharacterPowerLevelRadarChart characterChartData={characterChartData} />
          <CharacterPowerLevelBarChart characterChartData={characterChartData} />
        </div>
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
