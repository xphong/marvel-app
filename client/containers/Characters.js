import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as CharactersActions from '../actions/CharactersActions';
import SearchCharacterForm from '../components/characters/SearchCharacterForm';
import CharactersList from '../components/characters/CharactersList';
import WithLoading from '../components/helpers/WithLoading';

const CharactersListWithLoading = WithLoading(CharactersList);

export class Characters extends Component {
  componentDidMount() {
    if (this.props.params.name && this.props.params.name !== '') {
      this.props.actions.fetchCharactersByName(this.props.params.name);
    }
  }

  render() {
    const { characters, actions } = this.props;

    return <div>
        <SearchCharacterForm actions={actions} />
        <CharactersListWithLoading isLoading={characters.isLoading} characters={characters} />
      </div>;
  }
}

Characters.propTypes = {
  characters: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    characters: state.characters
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(CharactersActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Characters);
