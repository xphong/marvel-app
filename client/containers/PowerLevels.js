import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as PowerLevelActions from '../actions/PowerLevelActions';
import PowerLevelList from '../components/powerlevels/PowerLevelList';
import WithLoading from '../components/hocs/WithLoading';

const PowerLevelListWithLoading = WithLoading(PowerLevelList);

export class PowerLevels extends Component {
  componentDidMount() {
    this.props.actions.fetchPowerLevels();
  }

  render() {
    const { powerLevels, actions } = this.props;

    return (
      <div className="fadeIn-page-animation">
        <PowerLevelListWithLoading isLoading={powerLevels.isLoading} powerLevels={powerLevels} actions={actions} />
      </div>
    );
  }
}

PowerLevels.propTypes = {
  powerLevels: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    powerLevels: state.powerLevels
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(PowerLevelActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PowerLevels);
