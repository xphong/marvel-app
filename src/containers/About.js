import React, { Component } from 'react';

export default class About extends Component {
  render() {
    return (
      <div className='fadeIn-page-animation'>
        <h1>About</h1>
        <h4>Marvel Search App</h4>
        <ul>
          <li>Searches the Marvel database for the characters, power levels</li>
          <li>Built using <a href='https://github.com/xphong/simple-react-redux-starter' target='_blank'>simple-react-redux-starter</a> and the <a href='http://developer.marvel.com/' target='_blank'>Marvel API</a></li>
          <li><a href='https://github.com/xphong/marvel-react-app' target='_blank'>Github Link</a></li>
          <li><a href='http://marvel.com' target='_blank'>Data provided by Marvel. Â© {new Date().getFullYear()} Marvel</a></li>
        </ul>
      </div>
    );
  }
}
