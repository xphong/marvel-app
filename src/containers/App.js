import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Spinner from '../components/Spinner';
import * as CountriesActions from '../actions/CountriesActions';
import Countries from '../components/Countries';

export class App extends Component {
  componentDidMount() {
    this.props.actions.fetchCountries('https://restcountries.eu/rest/v1/all');
  }

  render() {
    const { countries, actions } = this.props;
    
    return (
      <div className="mainContainer container">
        <h1>React Redux Starter</h1>
        {countries.isLoading ? <Spinner /> : <Countries countries={countries} actions={actions} />}
      </div>
    );
  }
}

App.propTypes = {
  countries: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    countries: state.countries
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(CountriesActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
