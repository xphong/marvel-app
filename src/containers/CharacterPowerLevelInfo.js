import React, { Component, PropTypes } from 'react';

import CharacterPowerLevelInfoDisplay from '../components/characterpowerlevels/CharacterPowerLevelInfoDisplay';
import CharacterPowerLevelRadarChart from '../components/characterpowerlevels/CharacterPowerLevelRadarChart';

export default class CharacterPowerLevelInfo extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    return {
      labels: ["Eating", "Drinking", "Sleeping", "Designing", "Coding", "Cycling", "Running"],
      datasets: [
        {
          label: "Character Data Set",
          data: [28, 48, 40, 19, 96, 27, 100]
        }
      ]
    };
  }

  render() {
    const {data} = this.props.powerLevels;
    const {id} = this.props;
    const character = data[id];

    if (!character) {
      return null;
    }

    const characterChartData = this.getCharacterChartData(character);

    return (
      <div className='ui container center aligned'>
        <CharacterPowerLevelInfoDisplay character={character} />
        <CharacterPowerLevelRadarChart characterChartData={characterChartData} />
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
