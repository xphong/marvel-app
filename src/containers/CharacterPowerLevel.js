import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Spinner from '../components/ui/Spinner';
import * as PowerLevelActions from '../actions/PowerLevelActions';
import CharacterPowerLevelInfo from '../components/characterpowerlevels/CharacterPowerLevelInfo';

class CharacterPowerLevel extends Component {
  componentDidMount() {
    if (!this.props.powerLevels.data.length) {
      this.props.actions.fetchPowerLevels();
    }
  }

  render() {
    const {powerLevels, params} = this.props;

    return (
      <div>
        {powerLevels.isLoading ? <Spinner /> : <CharacterPowerLevelInfo id={params.id} powerLevels={powerLevels} />}
      </div>
    );
  }
}

CharacterPowerLevel.propTypes = {
  powerLevels: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    powerLevels: state.powerLevels
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(PowerLevelActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CharacterPowerLevel);
