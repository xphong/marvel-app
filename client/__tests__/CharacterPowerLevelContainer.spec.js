import React from 'react';
import ReactDOM from 'react-dom';

import { CharacterPowerLevel } from '../containers/CharacterPowerLevel';

function setup({ isLoading = false, id = 1 } = {}) {
  const props = {
    actions: {
      fetchPowerLevels: jest.fn()
    },
    powerLevels: {
      isLoading,
      data: []
    },
    params: {
      id
    }
  };

  const wrapper = shallow(<CharacterPowerLevel {...props} />);

  return {
    props,
    wrapper
  };
}

describe('CharacterPowerLevel Container', () => {
  it('should render CharacterPowerLevel', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CharacterPowerLevel with loading spinner', () => {
    const { wrapper } = setup({
      isLoading: true
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchPowerLevels', () => {
    const { props } = setup();

    expect(props.actions.fetchPowerLevels).toHaveBeenCalled();
  });
});
