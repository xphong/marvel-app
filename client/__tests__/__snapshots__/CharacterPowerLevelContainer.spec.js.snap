// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPowerLevel Container should render CharacterPowerLevel 1`] = `
<div>
  <Component
    id={1}
    isLoading={false}
    powerLevels={
      Object {
        "data": Array [],
        "isLoading": false,
      }
    }
  />
</div>
`;

exports[`CharacterPowerLevel Container should render CharacterPowerLevel with loading spinner 1`] = `
<div>
  <Component
    id={1}
    isLoading={true}
    powerLevels={
      Object {
        "data": Array [],
        "isLoading": true,
      }
    }
  />
</div>
`;
